shared/src/main/scala/scala/util/parsing/combinator/PackratParsers.scala

```diff
@@ -107,7 +107,7 @@ trait PackratParsers extends Parsers {
     val q = super.phrase(p)
     new PackratParser[T] {
       def apply(in: Input) = in match {
-        case in: PackratReader[_] => q(in)
+        case in if in.isInstanceOf[PackratReader[_]] => q(in)
```

Comment:
interesting - can you explain this?

Reply:
As I read it, dotty doesn't infer that `PackratReader` is an `Input` leading to a `Type Mismatch Error`.

Reply:
it looks like https://github.com/lampepfl/dotty/pull/8413 fixed this for the next Dotty version comes, so I suggest adding a TODO comment saying so

Reply:
I tried dotty 0.23.0-RC1 (just released) and that indeed fixes this issue.